### ISSUE TITLE
Launcher doesn't work with Java 16

### ISSUE THREAD

**Issue by RyanTheAllmighty** [ATLauncher/ATLauncher#465]
When opening the launcher with Java 16, instances fail to load with the below error:

```
[21/04/2021 08:58:14 AM] java.lang.reflect.InaccessibleObjectException: Unable to make field int java.awt.Color.value accessible: module java.desktop does not "opens java.awt" to unnamed module @5ffead27
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:177)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:171)
	at com.google.gson.internal.reflect.UnsafeReflectionAccessor.makeAccessible(UnsafeReflectionAccessor.java:44)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:159)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory.create(CollectionTypeAdapterFactory.java:53)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.Gson.fromJson(Gson.java:931)
	at com.google.gson.Gson.fromJson(Gson.java:870)
	at com.atlauncher.managers.InstanceManager.loadInstances(InstanceManager.java:67)
	at com.atlauncher.Launcher.loadEverything(Launcher.java:125)
	at com.atlauncher.App.main(App.java:354)
```

**Comment by RyanTheAllmighty** [comment]
Looks to be an issue with the Gson library which hasn't been updated since 2019. Moving to another library like Jackson is going to be much too complex

**Comment by dgelessus** [comment]
Haven't looked into this in detail, but wouldn't it work to register a custom Gson type adapter for `java.awt.Color` that de-/serializes the object using public APIs? (`Color.value` field can be read using `Color.getRGB()` and can be set on creation using `new Color(value, true)`.) That should avoid using Gson's default reflection-based behavior which is causing the error here.

**Comment by RyanTheAllmighty** [comment]
> Haven't looked into this in detail, but wouldn't it work to register a custom Gson type adapter for `java.awt.Color` that de-/serializes the object using public APIs? (`Color.value` field can be read using `Color.getRGB()` and can be set on creation using `new Color(value, true)`.) That should avoid using Gson's default reflection-based behavior which is causing the error here.

I never even thought about that 🤦🏻

**Comment by RyanTheAllmighty** [comment]
Released with version 3.4.4.0 of the launcher
